src/backend/api/meals.js

```diff
@@ -1,12 +1,79 @@
 const express = require("express");
+const { request, response } = require("../app");
 const router = express.Router();
 const knex = require("../database");
 
+// maxPrice
 router.get("/", async (request, response) => {
+  let meals = knex("meal");
+  if ("maxPrice" in request.query) {
+    if (!"maxPrice") {
+      response.status(403).json("Please enter data");
+    } else {
+      meals = meals.where("price", "<=", Number(request.query.maxPrice));
+    }
+  }
+  // availableReservations
+
+  if ("availableReservation" in request.query) {
+    const reservation = request.query.availableReservation;
+    if (false(reservation) && typeof reservation == "boolean") {
+      response.status("Sorry there are no reservations!");
+    } else if (true(reservation) && typeof reservation == "boolean") {
+      meals = meals
+        .join("reservation", "meal.id", "=", "reservation.meal_id")
+        .select(
+          "meal.id",
+          "title",
+          "max_reservations",
+          knex.raw("(meal.max_reservations-SUM(reservation.number_of_guests)) ")
+        )
+        .where("meal.max_reservations", ">", "reservation.number_of_guests")
+        .groupBy("meal.id")
+        .having(
+          "(meal.max_reservations-SUM(reservation.number_of_guests)) > 0"
+        );
```

Comment:
Does this return the available reservations? I have tried a similar method but it doesn't work and I change some other way.